lib/engine/part/destination.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Part
+    class Destination < Base
+      # Functionally is an extension of Icon
```

Comment:
but you don't extend icon?

can we remove destination and just use icon? what does 18AL do?

Reply:
It extends Icon now
The reason behind creating a Destination part instead of just using Icon is so that the Destination part can be used for logic, particularly for 1856 escrow-release logic

Reply:
that would be a reason not to have a destitanion part, having logic in the models makes it less extensible unless the logic is super generic

Reply:
`Destination`s as they are here are generic - the logic would only live in the game / step code.
For example, 18FL may override `calculate_revenue`to increase the value of towns which a corporation has a hotel (`Destination`) in
1856 will use them to represent the hexes to connect on the map and have them visually toggle-able
1822 and 1870 can have them on the map to indicate destinations for corporations and remove them once the corporation connects and places its station in the city 
18AL can also represent them on the map and could have implemented the HIstorical Objective logic by checking if the hex had the corporation's destination on it in the Token step

This way in the model, a `Destination` is simply an `Icon` that:
 * is associated with a corporation or a minor
 * can be toggled on or off in the map display independently of other Icons
 
 If you're OK with leaving `Destination`s as its own Part, something I want to do next in this PR is:
 Add a method to Game that lets a game decide what they are called (18FL: hotel; 18AL: Historical Objective; default: Destination)
 Add methods in `Game::Base` to add and remove destinations
 Add them to 18AL's config as an example of them being able to be added that way

Reply:
I don't want the destinations to be renamed just like i don't want corporations to be renamed to share companies. Everything should just be called a destination.

Reply:
Destination as a Part is gone; it was removed in a single commit so it could be revived if wanted later.